pkg/gui/mergeconflicts/state.go

```diff
@@ -98,22 +98,19 @@ func findConflicts(content string) []*mergeConflict {
 	var newConflict *mergeConflict
 	for i, line := range utils.SplitLines(content) {
 		trimmedLine := strings.TrimPrefix(line, "++")
-		switch trimmedLine {
-		case "<<<<<<< HEAD", "<<<<<<< MERGE_HEAD", "<<<<<<< Updated upstream", "<<<<<<< ours":
+
+		if strings.HasPrefix(trimmedLine, "<<<<<<< ") {
```

Comment:
At this point I'm wondering if we should just update the switch to look for [true statements](https://tour.golang.org/flowcontrol/11). Ie, something like:

```go
switch {
	strings.HasPrefix(trimmedLine, "<<<<<<< "):
		blah
	trimmedLine == "":
		blah
	strings.HasPrefix(trimmedLine, ">>>>>>> "):
		blah
}
```

thoughts?

Reply:
I actually just merged a refactor PR I made on the back of your PR @harrisonhjones: https://github.com/jesseduffield/lazygit/pull/1347

That uses a regex-based approach, but the above snippet seems simpler and less error-prone. The reason I initially whitelisted things like HEAD and 'Updated upstream' was so that I wasn't miscategorising non-conflict markers in code, but I'm happy to switch gears and see if we run into any trouble just checking for '<<<<<<< ' and '>>>>>>> '

Reply:
If we run into any issues with people actually having '<<<<<<< ' and '>>>>>>> ' in their code we can cross that bridge when we come to it.

Reply:
Thanks @harrisonhjones , I think the switch block is appropriate.
@jesseduffield I have merged your changes from the master branch and applied my approach to it. I have also added a `newConflict = nil` to avoid silent mutations of previous objects (if the `START` case does not succeed). I'd go with "loud failures" which might be easier to trace where it failed. Let me know if you disagree with anything.

> If we run into any issues with people actually having '<<<<<<< ' and '>>>>>>> ' in their code we can cross that bridge when we come to it.

Indeed. I did try to find any "official" info about the uniqueness of the markers. The closest I got is a contrib tool in the `git` repo that also just checks the `<<<<<<<` prefix only: https://github.com/git/git/blob/master/contrib/git-jump/git-jump#L46.